### PR TITLE
Apply inline start padding to selected lines on modern layout only

### PR DESCRIPTION
**Pull request by luixxiul**
Fixes https://github.com/vector-im/element-web/issues/22768

This PR removes inline start padding from selected lines on IRC layout and applies it to ones on modern layout.

|Before|After|
|---------|------|
|before1|after|

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

type: defect

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Apply inline start padding to selected lines on modern layout only ([\#9006](https://github.com/matrix-org/matrix-react-sdk/pull/9006)). Fixes vector-im/element-web#22768. Contributed by @luixxiul.

### CODE REVIEW

**Comment by kerryarchibald** [comment]
The expand/collapse icon button on hidden events is overflowing outside the message highlight in IRC

Develop:
Screenshot 2022-07-07 at 14 29 46

netlify:
Screenshot 2022-07-07 at 14 29 59

**Comment by kerryarchibald** [comment]
@luixxiul We would really like to start covering some of these sensitive areas with screenshot tests to give us more confidence in changes. 
We use percy along with cypress for screenshot tests, our [cypress docs are here](https://github.com/matrix-org/matrix-react-sdk/blob/develop/docs/cypress.md), this section especially about [screenshots](https://github.com/matrix-org/matrix-react-sdk/blob/develop/docs/cypress.md#percy-visual-testing). 
The [threads tests](https://github.com/matrix-org/matrix-react-sdk/blob/develop/cypress/e2e/5-threads/threads.spec.ts#L81-L92) already create a room and send a message so could be a good jumping off point. 

Would you feel comfortable adding a small test case to cover some of the scenarios here?

File: cypress/e2e/14-timeline/timeline.spec.ts [hunk 1]
```diff
@@ -141,5 +142,40 @@ describe("Timeline", () => {
                 expectAvatar(e, newAvatarUrl);
             });
         });
+
+        it("should select a hidden event line on IRC layout", () => {
+            cy.setSettingValue("useOnlyCurrentProfiles", null, SettingLevel.ACCOUNT, false);
```
Comment (SimonBrandner) [inline cypress/e2e/14-timeline/timeline.spec.ts:147]: I think this is a leftover that doesn't need to be here?

File: cypress/e2e/14-timeline/timeline.spec.ts [hunk 2]
```diff
@@ -141,5 +142,40 @@ describe("Timeline", () => {
...
+            // User sends message
+            cy.get(".mx_RoomView_body .mx_BasicMessageComposer_input").type("Hello{enter}");
+
+            // Wait for message to send
+            cy.get(".mx_RoomView_body .mx_EventTile").contains(".mx_EventTile[data-scroll-tokens]", "Hello");
+
+            // Edit message
+            cy.get(".mx_RoomView_body .mx_EventTile").contains(".mx_EventTile_line", "Hello").within(() => {
+                cy.get('[aria-label="Edit"]').click({ force: true }); // Cypress has no ability to hover
+                cy.get(".mx_BasicMessageComposer_input").type(" Edit{enter}");
```
Comment (SimonBrandner) [inline cypress/e2e/14-timeline/timeline.spec.ts:168]:
```suggestion
                cy.get(".mx_BasicMessageComposer_input").type("Edit{enter}");
```

File: cypress/e2e/14-timeline/timeline.spec.ts [hunk 3]
```diff
@@ -141,5 +142,40 @@ describe("Timeline", () => {
...
+            cy.setSettingValue("useOnlyCurrentProfiles", null, SettingLevel.ACCOUNT, false);
+
+            // Enable hidden events and IRC layout
+            cy.setSettingValue("showHiddenEventsInTimeline", null, SettingLevel.DEVICE, true);
+            cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.IRC);
+
+            cy.visit("/#/room/" + roomId);
+
+            // Wait until configuration is finished
+            cy.get(".mx_RoomView_body .mx_GenericEventListSummary[data-layout=irc] .mx_GenericEventListSummary_summary")
```
[the remark this answers is not in the document]
Comment (luixxiul) [reply cypress/e2e/14-timeline/timeline.spec.ts:156]: Thanks for the information. This is my very first time to write a test, so any suggestion helps!

File: res/css/views/rooms/_EventTile.scss [hunk 4]
```diff
@@ -409,6 +409,10 @@ $left-gutter: 64px;
                     top: 0;
                 }
             }
+
+            &.mx_EventTile_selected .mx_EventTile_line {
```
[the remark this answers is not in the document]
Comment (luixxiul) [reply res/css/views/rooms/_EventTile.scss:413]: This PR intends to add inline start padding (padding-left) only to the modern layout, removing it from IRC layout, so what you suggest is to create another same test for the modern layout in order to make sure that a regression does not happen on that layout? :thinking:

File: cypress/e2e/14-timeline/timeline.spec.ts (hunk 2 above)
Comment (SimonBrandner) [reply cypress/e2e/14-timeline/timeline.spec.ts:168]: Ah, probably doesn't really matter

File: res/css/views/rooms/_EventTile.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (luixxiul) [reply res/css/views/rooms/_EventTile.scss:413]: Created another test for modern/group layout.

**Comment by luixxiul** [comment]
What would be a preferred way of waiting until room configuration is finished? `cy.contains(".mx_RoomView_body .mx_GenericEventListSummary[data-layout=irc] " + ...` seems to fail intermittently.

**Comment by luixxiul** [comment]
Here is the cropped screenshot of the cypress run on CI.

Untitled

The `Hello` line is not supposed to appear at the top of the room.

**Comment by luixxiul** [comment]
Creating a test for modern/group layout, I found an inconsistent padding setting of selected info event line. The commit above should fix it.

|Before|After|
|---------|------|
|before|after|

**Review by weeman1337 — CHANGES_REQUESTED** [review]

**Review by SimonBrandner — APPROVED** [review]
Looks sane

**Review by weeman1337 — APPROVED** [review]
:+1:

**Pull request merged**